app/src/main/java/com/keylesspalace/tusky/components/compose/dialog/FocusDialog.kt

```diff
@@ -0,0 +1,235 @@
+import android.os.Build
+import android.view.WindowManager
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import at.connyduck.sparkbutton.helpers.Utils
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
+import com.bumptech.glide.load.resource.bitmap.DownsampleStrategy
+import com.bumptech.glide.load.resource.bitmap.TransformationUtils
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import com.bumptech.glide.util.Util
+import com.github.chrisbanes.photoview.OnPhotoTapListener
+import com.github.chrisbanes.photoview.PhotoView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.entity.Attachment.Focus
+import kotlinx.coroutines.launch
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+import java.util.concurrent.locks.Lock
+
+// Private, but necessary to implement BitmapTransformation, function extracted from Glide
+private fun getAlphaSafeBitmap(pool: BitmapPool, maybeAlphaSafe: Bitmap): Bitmap {
+    val safeConfig: Bitmap.Config = getAlphaSafeConfig(maybeAlphaSafe)
+    if (safeConfig == maybeAlphaSafe.config) {
+        return maybeAlphaSafe
+    }
+    val argbBitmap = pool[maybeAlphaSafe.width, maybeAlphaSafe.height, safeConfig]
+    Canvas(argbBitmap).drawBitmap(maybeAlphaSafe, 0.0f /*left*/, 0.0f /*top*/, null /*paint*/)
+
+    // From Glide: "We now own this Bitmap. It's our responsibility to replace it in the pool outside this method
+    // when we're finished with it."
+    return argbBitmap
+}
+
+// Private, but necessary to implement BitmapTransformation, function extracted from Glide
+private fun getAlphaSafeConfig(inBitmap: Bitmap): Bitmap.Config {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        // Avoid short circuiting the sdk check.
+        if (Bitmap.Config.RGBA_F16 == inBitmap.config) { // NOPMD
+            return Bitmap.Config.RGBA_F16
+        }
+    }
+    return Bitmap.Config.ARGB_8888
+}
+
+private val transparentDarkGray = 0x40000000
+
+/** Glide BitmapTransformation which overlays a highlight on a focus point. */
+class HighlightFocus(val focus: Focus) : BitmapTransformation() {
+    override fun transform(pool: BitmapPool, inBitmap: Bitmap, outWidth: Int, outHeight: Int): Bitmap {
+        // Draw overlaid target
+        val bitmapDrawableLock: Lock = TransformationUtils.getBitmapDrawableLock()
+        val safeConfig: Bitmap.Config = getAlphaSafeConfig(inBitmap)
+        val toTransform: Bitmap = getAlphaSafeBitmap(pool, inBitmap)
+        val result = pool[toTransform.width, toTransform.height, safeConfig]
+
+        val plainPaint = Paint()
+        val strokePaint = Paint(Paint.ANTI_ALIAS_FLAG)
```

Comment:
Very unclear if ANTI_ALIAS_FLAG is required, I tried it both ways and I THINK it looks better with but it's hard to tell